app/controllers/admin/pages_controller.rb

```diff
@@ -67,6 +67,7 @@ def create
           page.save!
         else
           page = Page.create!(image: image)
```

Comment:
What you did works fine in theory, but it will not actually persist the property value to the DB. On line 69, the page is being initialized and saved in one step. When you set the `visible` property on line 70, you are setting it on the object but not saving it to the DB.

What I would suggest that you do instead is to add the `visible` property to the `Page.create!` method call on line 69 as another keyword argument. This would look like `page = Page.create!(image: image, visible: false)`. This way, the visible property is set and saved at the same time that the page image is saved.